docs/markup-language/README.md

```diff
@@ -340,6 +340,58 @@ If either value is not defined at the time it is to be rendered in the draft or
 
 The OLInfo type will be populated with additional sub-fields in the future.
 
+### External Signature
+
+The ExternalSignature type lets you define a new signature method which uses an external service registered in Openlaw platform through Integrator's API.
+The following is an example of the syntax to define and use an eletronic signature using DocuSign external service:
+
+```
+[[Signatory: ExternalSignature(serviceName:"DocuSign")]]
+```
+
+### External Call
+
+The ExternalCall type allows you to define custom calls that should be triggered from Openlaw VM to an external service.
+In the following example we define an external call to Coin Market Cap service to get an exchange rate of BTC in USD for a certain amount.
+
```

Comment:
I think it is good but I am wondering if we should also give a more generic example as well. It might be hard for someone to integrate their own custom service without this

Reply:
Right. Do you have any example in mind?

Reply:
hmm, maybe something like:

```
[[Input: Structure(param1:Type, param2:Type)]]
[[Output: Structure(param1:Type, param2:Type)]]
```

and then referencing the more generic stuff below

Reply:
sounds god, I will update that. Thanks